server/utils/utils.ts

```diff
@@ -81,5 +81,5 @@ export function moveGoal(goals: Array<any>, uuid: string, operation: string) {
   ;[orderedGoals[index], orderedGoals[targetIndex]] = [orderedGoals[targetIndex], orderedGoals[index]]
   orderedGoals[index].goalOrder = index
   orderedGoals[targetIndex].goalOrder = targetIndex
```

Comment:
So I _think_ this is OK but because Goals can be deleted and we don't change the goal order of the other goals in the plan when that happens, the initial goalOrder values and index position in orderedGoals won't necessarily be the same. Are there any edge cases we need to be aware of, or is it fine as-is?

Reply:
Dont think any issue with this as it only reverse the 2 goal order. May be on remove goal, we could reorder all goals again but this will not be impacted.